sdk/core/core-client/src/utils.ts

```diff
@@ -82,30 +146,34 @@ export function flattenResponse(
           arrayResponse[key] = parsedHeaders[key];
         }
       }
-      return arrayResponse;
+      return !fullResponse.parsedBody && isNullable && arrayResponse === emptyArray
+        ? null
+        : arrayResponse;
     }
 
     if (typeName === "Composite" || typeName === "Dictionary") {
-      return {
-        ...parsedHeaders,
-        ...fullResponse.parsedBody
-      };
+      return handleNullableResponseAndWrappableBody({
```

Comment:
Seems like this whole if branch can be pruned to use the fallback below, no?

Reply:
You're right! Much cleaner now, thanks!